python/pyspark/serializers.py

```diff
@@ -208,8 +214,26 @@ def load_stream(self, stream):
         for batch in reader:
             yield batch
 
+        if self.load_batch_order:
+            num = read_int(stream)
+            self.batch_order = []
```

Comment:
If we're going to have get_batch_order_and_reset as a separate function, could we verify batch_order is None before we reset and throw here if it's not? Just thinking of future folks who might have to debug something here.